### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ihorandrianov**
[DEMO LINK](https://ihorandrianov.github.io/react_static-list-of-posts/)

### CODE REVIEW

**Review by MaxManis — APPROVED** [review]
Nice!)

File: src/components/PostList/PostList.tsx [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
-);
+type PostsProps = {
+  posts: Post[];
+};
+
+export const PostList: React.FC<PostsProps> = ({ posts }) => {
+  return (
+    <div className="PostList">
+      {posts.map((post) => (<PostInfo post={post} />))}
+    </div>
```
Comment (OleksiiBlezniuk) [inline src/components/PostList/PostList.tsx:13]: Let's use semantic tags for list and it's items
Also don't forget about `key` attribute (it should be in parent component, not in child)

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

File: src/components/CommentInfo/CommentInfo.tsx [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
-  <>Put the comment here</>
-);
+type CommentProps = {
+  comment: Comment;
+};
+
+export const CommentInfo: React.FC<CommentProps> = ({ comment }) => {
+  const {
+    name, email, body,
+  } = comment;
```
Comment (silchyn) [inline src/components/CommentInfo/CommentInfo.tsx:11]:
```suggestion
  const {
    name,
    email,
    body,
  } = comment;
```

File: src/components/PostInfo/PostInfo.tsx [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
-  <>Put the post here</>
-);
+type PostProps = {
+  post: Post
+};
+
+export const PostInfo: React.FC<PostProps> = ({ post }) => {
+  const {
+    id, title, body, user, comments,
+  } = post;
```
Comment (silchyn) [inline src/components/PostInfo/PostInfo.tsx:13]:
```suggestion
  const {
    id,
    title,
    body,
    user,
    comments,
  } = post;
```

File: src/components/PostInfo/PostInfo.tsx [hunk 4]
```diff
@@ -1,5 +1,38 @@
...
+        {body}
+      </p>
+      {comments.length > 0
+        ? <CommentList comments={comments} />
+        : (
+          <>
+            <hr />
+            <b data-cy="NoCommentsMessage">No comments yet</b>
+          </>
+        )}
```
Comment (silchyn) [inline src/components/PostInfo/PostInfo.tsx:35]:
```suggestion
      <p className="PostInfo__body">
        {body}
      </p>
      
      {comments.length > 0
        ? <CommentList comments={comments} />
        : (
          <>
            <hr />
            <b data-cy="NoCommentsMessage">No comments yet</b>
          </>
        )}
```

File: src/components/PostList/PostList.tsx (hunk 1 above)
Comment (silchyn) [inline src/components/PostList/PostList.tsx:13]: > Let's use semantic tags for list and it's items. Also don't forget about `key` attribute (it should be in parent component, not in child)

File: src/components/UserInfo/UserInfo.tsx [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
-  <>Put the user here</>
-);
+type UserProps = {
+  user: User;
+};
+
+export const UserInfo: React.FC<UserProps> = ({ user }) => {
+  const {
+    name, username, email,
+  } = user;
```
Comment (silchyn) [inline src/components/UserInfo/UserInfo.tsx:11]:
```suggestion
  const {
    name,
    username,
    email,
  } = user;
```

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/components/CommentInfo/CommentInfo.tsx [hunk 6]
```diff
@@ -1,5 +1,33 @@
...
+
+  return (
+    <li className="CommentInfo">
+      <div className="CommentInfo__title">
+        <strong className="CommentInfo__name">{name}</strong>
+
+        {' by '}
```
Comment (naziamoff) [inline src/components/CommentInfo/CommentInfo.tsx:18]:
```suggestion
        <strong className="CommentInfo__name">
          {name}
        </strong>
        {' by '}
```

File: src/components/PostInfo/PostInfo.tsx [hunk 7]
```diff
@@ -1,5 +1,39 @@
...
+        </h3>
+
+        {user && <UserInfo user={user} />}
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      {comments.length > 0
```
Comment (naziamoff) [inline src/components/PostInfo/PostInfo.tsx:29]: We can keep it simple with boolean
```suggestion
      {comments.length
```

**Review by naziamoff — APPROVED** [review]
GJ

**Pull request closed without merging**